Objects/descrobject.c

```diff
@@ -2112,7 +2142,7 @@ ga_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 // - __eq__
 PyTypeObject Py_GenericAliasType = {
     PyVarObject_HEAD_INIT(&PyType_Type, 0)
-    .tp_name = "GenericAlias",
+    .tp_name = "types.GenericAlias",
```

Comment:
Hopefully this is okay, pickle would try to import `GenericAlias` from builtins otherwise.

Reply:
Yes, indeed this seems how it's done, see e.g. odictobject.c:1550, picklebufobject.c:208, namespaceobject.c:206 (all in Objects).